File: learn/math/probability/birthday.py
import numpy as np

K = np.array(range(0, 101))
Y = np.array([])

days = 365
day = days
perm = 1

for k in K:
    day = days
    while day >= days - k + 1:
        perm *= day
        day -= 1
    y = 1 - ((perm) // days ** k)
    Y = np.append(Y, y)

print(perm)
print(Y)

File: learn/math/probability/birthday.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main(void)
{
    int K[10] = { 0 };
    double Y[10] = { 0 };
    int days = 365;
    int day = days;
    unsigned long perm = 1;
    long double y = 0;

    for (int i = 0; i < 10; i++)
        K[i] = i + 1;

    for (int i = 0, k = 0; i < 10; i++, day = days, y = 0)
    {
        k = K[i];
        while (day > (days - k + 1))
        {
            perm *= day--;
        }
        y = (long double)(1) - ((long double)(perm) / pow(days, (double)k));
        Y[i] = y;
    }

    cout << perm << endl;
    for (int i = 0; i < 10; i++)
    {
        cout << K[i] << " " << Y[i] << endl;
    }

    return 0;
}